server/repository/src/db_diesel/master_list_row.rs

```diff
@@ -52,22 +52,12 @@ impl<'a> MasterListRowRepository<'a> {
         Ok(())
     }
 
-    pub async fn find_one_by_id(
-        &self,
-        master_list_id: &str,
-    ) -> Result<MasterListRow, RepositoryError> {
-        let result = master_list
-            .filter(id.eq(master_list_id).and(is_active.eq(true)))
-            .first(&self.connection.connection)?;
-        Ok(result)
-    }
-
-    pub fn find_one_by_id_option(
+    pub fn find_one_by_id(
         &self,
         master_list_id: &str,
     ) -> Result<Option<MasterListRow>, RepositoryError> {
         let result = master_list
-            .filter(id.eq(master_list_id).and(is_active.eq(true)))
+            .filter(id.eq(master_list_id))
```

Comment:
I should have added this as a seperate issue/pr, sorry. Or at the very least explained what is going on with this change:

I've noticed we are not actually checking `inactive` -> `is_active` in unit tests for sync:

https://github.com/msupply-foundation/open-msupply/blob/23697a8701506c95892bcef42a22df54a27782fa/server/service/src/sync/test/test_data/master_list.rs#L24

https://github.com/msupply-foundation/open-msupply/blob/23697a8701506c95892bcef42a22df54a27782fa/server/service/src/sync/test/test_data/master_list.rs#L44

And since this field was missing in integration tests, when i added `inactive` = true variation, the tests were failing since the confirmation is done by find_one_by_id (which is always ignoring non active). Find by reference of find_one_by_id showed it's only used in tests, so decided it was safe to remove is_active.

Btw what do you think about this in general. If we soft deleting should we auto filter out in row repo or "domain" repo ?

Reply:
Sorry about the failing test, i did mention i fixed them, and they must have worked at some point before PR, I should have checked again, fix:

3e9603e3a2a75d2c7edb29256efbd5f15c7e5017

Reply:
> Btw what do you think about this in general. If we soft deleting should we auto filter out in row repo or "domain" repo ?

I would say, filter out on default in the domain repo but have the option to fetch the deleted entries.

Maybe for an offline discussion: do we need two repos? For the rename it would have come in handy if the `MyDieselRow` object would be private to the repo, i.e. only have a public row object that has the actual name_id/item_id instead of name_link_id/item_link_id...

Reply:
> I would say, filter out on default in the domain repo but have the option to fetch the deleted entries.

Agree. Although we still need 'fine_one_by_id' right ?

Yeah, keen to chat about this offline, quick thought:. For now we need the MyDeiselRow because we save the whole row at once ?

 I think in the case of name_id/item_id, if those are missing from MyDieselRow, that might have been better ? Have another method and dsl to update MyDieselRow for those, can even do an inner statement to find item_id/name_id ?

Reply:
Yes something in this line, I sneaked in the idea already here:
https://github.com/msupply-foundation/open-msupply/pull/2892

but will try to catch up with you today